packages/x-scheduler/src/material/timeline/content/view-header/TimeHeader.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import { useStore } from '@base-ui-components/utils/store/useStore';
+import { getAdapter } from '../../../../primitives/utils/adapter/getAdapter';
+import { useDayList } from '../../../../primitives/use-day-list';
+import { selectors } from '../../../../primitives/use-timeline';
+import { useTimelineStoreContext } from '../../../../primitives/utils/useTimelineStoreContext';
+
+import './Headers.css';
+
+const adapter = getAdapter();
+
+export function TimeHeader() {
+  const getDayList = useDayList();
+  const store = useTimelineStoreContext();
+
+  const visibleDate = useStore(store, selectors.visibleDate);
+  const ampm = useStore(store, selectors.ampm);
+
+  const timeFormat = ampm ? 'hoursMinutes12h' : 'hoursMinutes24h';
+
+  const days = React.useMemo(
+    () =>
+      getDayList({
+        date: visibleDate,
+        amount: 3,
+      }),
+    [getDayList, visibleDate],
+  );
+
+  return (
+    <div className="TimeHeader">
+      {days.map((day) => (
+        <div key={adapter.format(day.value, 'keyboardDate')} className="TimeHeaderCell">
+          <div className="DayLabel">{adapter.format(day.value, 'normalDateWithWeekday')}</div>
+          <div className="TimeCellsRow">
+            {Array.from({ length: 24 }, (_, hour) => (
```

Comment:
```suggestion
            // TODO: Make sure it works across DST
            {Array.from({ length: 24 }, (_, hour) => (
```

I suspect that we need to build the hour list for the real day using `addHours` 😢